### PR TITLE
portal.imageUrl renders confusing error #8583

### PR DESCRIPTION
**Pull request by vbradnitski**

### CODE REVIEW

File: modules/web/web-api/src/main/java/com/enonic/xp/web/WebException.java [hunk 1]
```diff
@@ -8,29 +8,46 @@
 {
     private final HttpStatus status;
 
+    private final boolean log;
```
Comment (rymsha) [inline modules/web/web-api/src/main/java/com/enonic/xp/web/WebException.java:11]: loggable
```suggestion
    private final boolean loggable;
```

File: modules/web/web-api/src/main/java/com/enonic/xp/web/WebException.java [hunk 2]
```diff
@@ -8,29 +8,46 @@
...
+
     public WebException( final HttpStatus status, final String message )
     {
         super( message );
         this.status = status;
+        this.log = true;
```
Comment (rymsha) [inline modules/web/web-api/src/main/java/com/enonic/xp/web/WebException.java:17]:
```suggestion
        this.loggable = true;
```

File: modules/web/web-api/src/main/java/com/enonic/xp/web/WebException.java [hunk 3]
```diff
@@ -8,29 +8,46 @@
...
+    }
+
+    public WebException( final HttpStatus status, final String message, final boolean log )
```
Comment (rymsha) [inline modules/web/web-api/src/main/java/com/enonic/xp/web/WebException.java:20]:
```suggestion
    public WebException( final HttpStatus status, final String message, final boolean loggable )
```

File: modules/web/web-api/src/main/java/com/enonic/xp/web/WebException.java [hunk 4]
```diff
@@ -8,29 +8,46 @@
...
+    {
+        super( message );
+        this.status = status;
+        this.log = log;
```
Comment (rymsha) [inline modules/web/web-api/src/main/java/com/enonic/xp/web/WebException.java:24]:
```suggestion
        this.loggable = loggable;
```

File: modules/web/web-api/src/main/java/com/enonic/xp/web/WebException.java [hunk 5]
```diff
@@ -8,29 +8,46 @@
...
     }
 
     public WebException( final HttpStatus status, final Throwable cause )
     {
         super( cause.getMessage(), cause );
         this.status = status;
+        this.log = true;
```
Comment (rymsha) [inline modules/web/web-api/src/main/java/com/enonic/xp/web/WebException.java:31]:
```suggestion
        this.loggable = true;
```

File: modules/web/web-api/src/main/java/com/enonic/xp/web/WebException.java [hunk 6]
```diff
@@ -8,29 +8,46 @@
...
     }
 
     public WebException( final HttpStatus status, final String message, final Throwable cause )
     {
         super( message, cause );
         this.status = status;
+        this.log = true;
```
Comment (rymsha) [inline modules/web/web-api/src/main/java/com/enonic/xp/web/WebException.java:38]:
```suggestion
        this.loggable = true;
```

File: modules/web/web-api/src/main/java/com/enonic/xp/web/WebException.java [hunk 7]
```diff
@@ -8,29 +8,46 @@
...
     }
 
     public HttpStatus getStatus()
     {
         return this.status;
     }
 
+    public boolean isLoggable()
+    {
+        return log;
```
Comment (rymsha) [inline modules/web/web-api/src/main/java/com/enonic/xp/web/WebException.java:48]:
```suggestion
        return loggable;
```

File: modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/exception/ExceptionInfo.java [hunk 8]
```diff
@@ -39,7 +40,7 @@ public HttpStatus getStatus()
 
     public boolean shouldLogAsError()
     {
-        return this.status.is5xxServerError() || this.status == HttpStatus.BAD_REQUEST;
+        return this.status.is5xxServerError() && cause.isLoggable();
```
Comment (rymsha) [inline modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/exception/ExceptionInfo.java:43]: Move this logic to WebException constructor

**Review by rymsha — APPROVED** [review]

File: modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/url/ImageUrlBuilder.java [hunk 9]
```diff
@@ -50,6 +52,12 @@ private void addParamIfNeeded( final Multimap<String, String> params, final Stri
 
     private Media resolveMedia( final ContentId id )
     {
+        final Content content = this.contentService.getById( id );
```
Comment (anatol-sialitski) [inline modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/url/ImageUrlBuilder.java:55]: At this line will be thrown an exception if node not found by ID

**Review by anatol-sialitski — APPROVED** [review]

**Pull request merged**